### PR TITLE
Use Babel in main process with Webpack build

### PR DESCRIPTION
**Pull request by jhen0409**
Related to #139, and just use `-r babel-core/register` in dev mode.

cc @chentsulin @davej

### CODE REVIEW

File: webpack.config.electron.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    { raw: true, entryOnly: false }
+  ),
+  new webpack.DefinePlugin({
+    'process.env': {
+      NODE_ENV: JSON.stringify('production')
+    }
+  })
+);
+config.externals.push('electron');
+config.target = 'node';
```
Comment (chentsulin) [inline webpack.config.electron.js]: should use 'electron' as target

Comment (jhen0409) [inline webpack.config.electron.js]: fixed :)

**Comment by jhen0409** [comment]
Also, maybe we can just pack externals of webpack config on `npm run package`?

File: webpack.config.production.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ config.entry = './app/index';
 
 config.output.publicPath = '../dist/';
 
-config.module.loaders.push({
+config.module.loaders = config.module.loaders.concat([{
```
Comment (jhen0409) [inline webpack.config.production.js]: Looks `.production.js` and `.electron.js` will have problem of object reference in `package.js`, so I handle these with `array.concat`.

**Comment by chentsulin** [comment]
I just wonder whether `babel-regiseter` is good enough for development..If we use some webpack specified feature in production, we can't directly apply them to development env.

File: package.json [hunk 3]
```diff
@@ -10,9 +10,11 @@
     "test-e2e": "cross-env NODE_ENV=test mocha --compilers js:babel-core/register --require ./test/setup.js --require co-mocha ./test/e2e.js",
     "lint": "eslint app test *.js",
     "hot-server": "babel-node server.js",
-    "build": "cross-env NODE_ENV=production babel-node ./node_modules/.bin/webpack --config webpack.config.production.js --progress --profile --colors",
+    "build-electron": "cross-env NODE_ENV=production babel-node ./node_modules/.bin/webpack --config webpack.config.electron.js --progress --profile --colors",
```
Comment (chentsulin) [inline package.json]: Are `build-main` and `build-renderer` nicer?

**Comment by jhen0409** [comment]
> If we use some webpack specified feature in production, we can't directly apply them to development env.

Yes, I can think of case is webpack loader, but it can be resolved by `babal-plugin-webpack-loaders`.

In addition, I recommend webpack build of main process is for production. @chentsulin What do you think?

**Comment by chentsulin** [comment]
I'm ok for that. I don't know why but there is a regression about #192. It seems window will open before content ready.

**Comment by jhen0409** [comment]
> It seems window will open before content ready.

It's `ERR_CONNECTION_REFUSED` error? It happens sometimes.

The `npm run hot-server` used `babel-node` (#195), so it's time to start may later than main process.

**Comment by chentsulin** [comment]
Could we somehow fix `ERR_CONNECTION_REFUSED` to improve user experience here?

**Comment by jhen0409** [comment]
Use `node -r babel-register` will be faster, it just not has some options and require `babel-polyfill`.

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
     "test": "cross-env NODE_ENV=test mocha --compilers js:babel-register --recursive --require ./test/setup.js test/**/*.spec.js",
     "test-watch": "npm test -- --watch",
     "test-e2e": "cross-env NODE_ENV=test mocha --compilers js:babel-register --require ./test/setup.js --require co-mocha ./test/e2e.js",
-    "lint": "eslint app test *.js",
+    "lint": "eslint app test *.js --ignore-pattern main.js",
```
Comment (chentsulin) [inline package.json]: Why not put this in a `.eslintignore`?

**Comment by chentsulin** [comment]
I am almost ready to merge this PR. @jhen0409 Could you move `--ignore-pattern` to a `.eslintignore`? Thanks a lot.

**Pull request merged**